sources/dataset.py

```diff
@@ -163,6 +139,73 @@ class (useful for semantic segmentation result printing)
                                      "aggregate": aggregate,
                                      "color": color}
 
+    def save(self, filename):
+        """Save dataset in a json file indicated by `filename`
+
+        Parameter
+        ---------
+        filename: object
+            String designing the relative path where the dataset must be saved
+        """
+        with open(filename, 'w') as fp:
+            json.dump({"image_size": self.image_size,
+                       "classes": self.class_info,
+                       "images": self.image_info}, fp)
+        utils.logger.info("The dataset has been saved into {}".format(filename))
+
+    def load(self, filename, nb_images=None):
+        """Load a dataset from a json file indicated by `filename`
+
+        Parameter
+        ---------
+        filename: object
+            String designing the relative path from where the dataset must be
+        loaded
+        nb_images: integer
+            Number of images that must be loaded (if None, the whole dataset is loaded)
+        """
+        with open(filename) as fp:
+            ds = json.load(fp)
+            self.image_size = ds["image_size"]
+            self.class_info = {int(k):ds["classes"][k] for k in ds["classes"].keys()}
+            if nb_images is None:
+                self.image_info = {int(k):ds["images"][k] for k in ds["images"].keys()}
```

Comment:
You can iterate on a dict without the `.keys()`. By default, when you iterate on a dict, you iterate on its keys.

```python
d = {"a": 12, "b": -7}
[x for x in d]
# equivalent result and better than
[x for x in d.keys()]
```

Reply:
That's true, and the code is even uselessly complicated, as the following seems enough:
```
self.class_info = ds["classes"]
```

Reply:
Oh no correction: I wrote this complicated statements as the goal is to convert dictionary keys as integers (instead of strings).